### PR TITLE
fix: [PL-38936]: show all tags in delegate listing page

### PR DESCRIPTION
**Pull request by namanharness**
### Summary

Issue:- Only 3 tags were shown in the delegate listing page for a delegate even though delegate contains more tags

Fix: Used the TagRenderer Component at Project page here as well to support the same UI for showing all tags 

#### Screenshots

https://github.com/harness/harness-core-ui/assets/110456765/dc416c72-e21e-46af-8ff7-01eab3fc0f38

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/27-platform/delegates/pages/delegates/DelegateListingItem.tsx [hunk 1]
```diff
@@ -75,12 +85,14 @@ const RenderDelegateName: Renderer<CellProps<DelegateGroupDetails>> = ({ row })
 const RenderTags: Renderer<CellProps<DelegateGroupDetails>> = ({ row }) => {
   const delegate = row.original
   const allSelectors = Object.keys(delegate.groupImplicitSelectors || {}).concat(delegate.groupCustomSelectors || [])
+  // TagsRenderer component accepts object { [key: string]: string } so converting the received array into acceptable format
+  const delegateTags = {} as { [key: string]: string }
+  allSelectors.forEach(item => (delegateTags[item] = ''))
```
Comment (namanharness) [inline src/modules/27-platform/delegates/pages/delegates/DelegateListingItem.tsx:80]: this manipulation is necessary 
code without this 
```
  const delegate = row.original
  const allSelectors = Object.keys(delegate.groupImplicitSelectors || {}).concat(delegate.groupCustomSelectors || [])
  // TagsRenderer component accepts object { [key: string]: string } so converting the received array into acceptable format
  // const delegateTags = {} as { [key: string]: string }
  // allSelectors.forEach(item => (delegateTags[item] = ''))
  return (
    delegate?.groupImplicitSelectors && (
      <>
        <Text lineClamp={1} margin={{ right: 'medium' }}>
          <TagsRenderer tags={allSelectors as any} length={3} />
        </Text>
      </>
    )
  )

```


https://github.com/harness/harness-core-ui/assets/110456765/435db6cb-3be3-4d85-8e66-7f32608e64f4

File: src/modules/27-platform/delegates/pages/delegates/DelegateListingItem.tsx [hunk 2]
```diff
@@ -75,12 +75,14 @@ const RenderDelegateName: Renderer<CellProps<DelegateGroupDetails>> = ({ row })
...
   return (
     delegate?.groupImplicitSelectors && (
       <>
         <Text lineClamp={1} margin={{ right: 'medium' }}>
-          <TagsViewer key="tags" tags={allSelectors.slice(0, 3)} />
-          <span key="hidenTags">{allSelectors.length > 3 ? '+' + (allSelectors.length - 3) : ''}</span>
+          <TagsRenderer tags={delegateTags} length={3} />
```
Comment (mayankVermaHarness) [inline src/modules/27-platform/delegates/pages/delegates/DelegateListingItem.tsx:85]: we have a check only on `groupImplicitSelectors` whereas `allSelectors` also depends on `groupCustomSelectors`. The check here should have been on both (legacy issue).

Let's check this separately if `groupCustomSelectors` can exist without `groupImplicitSelectors`, and if yes, let's replace the check here as such
 
 ```
 ...
 
 Object.keys(delegateTags).length > 0 ? (
      <>
      <Text lineClamp={1} margin={{ right: 'medium' }}>
         ...

      ) : null
      
 ```

**Review by mayankVermaHarness — APPROVED** [review]

File: src/modules/27-platform/delegates/pages/delegates/DelegateListingItem.tsx (hunk 2 above)
Comment (namanharness) [reply src/modules/27-platform/delegates/pages/delegates/DelegateListingItem.tsx:85]: when creating a delegate without any tag , under the `groupImplicitSelectors` a tag is already present which is equal to the name of delegate.

inside `groupCustomSelectors` it contains user added tags 

so i think we can use your code which resolves tag on resulting tagValues object

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/27-platform/delegates/pages/delegates/DelegateListingItem.tsx [hunk 3]
```diff
@@ -75,16 +75,16 @@ const RenderDelegateName: Renderer<CellProps<DelegateGroupDetails>> = ({ row })
...
-      <>
-        <Text lineClamp={1} margin={{ right: 'medium' }}>
-          <TagsViewer key="tags" tags={allSelectors.slice(0, 3)} />
-          <span key="hidenTags">{allSelectors.length > 3 ? '+' + (allSelectors.length - 3) : ''}</span>
-        </Text>
-      </>
-    )
-  )
+  // TagsRenderer component accepts object { [key: string]: string } so converting the received array into acceptable format
+  const delegateTags = {} as { [key: string]: string }
```
Comment (pankajsingh-harness) [inline src/modules/27-platform/delegates/pages/delegates/DelegateListingItem.tsx:79]:
```suggestion
  const delegateTags = allSelectors.reduce((tags, selector) => {
    tags[selector]= ''
    return tags
}, {})
```

**Review by pankajsingh-harness — COMMENTED** [review]
NIT: Added comment to do processing in a more functional way that an iterative way

**Review by mayankVermaHarness — APPROVED** [review]

**Pull request merged**